pkg/csi/service/vanilla/controller.go

```diff
@@ -1917,65 +1886,63 @@ func (c *controller) createFileVolume(ctx context.Context, req *csi.CreateVolume
+				fsEnabledClusterToDsInfoMap = c.authMgrs[vcHost].GetFsEnabledClusterToDsMap(ctx)
+			} else {
+				fsEnabledClusterToDsInfoMap = c.authMgr.GetFsEnabledClusterToDsMap(ctx)
+			}
 
-		if len(filteredDatastores) == 0 {
-			// when len(filteredDatastore)==0, it means vsan file service is not enabled on any vsan cluster
-			return nil, csifault.CSIVSanFileServiceDisabledFault, logger.LogNewErrorCode(log, codes.FailedPrecondition,
-				"no datastores found to create file volume, vsan file service may be disabled")
-		}
-		if multivCenterCSITopologyEnabled {
-			vc, err := c.managers.VcenterManager.GetVirtualCenter(ctx, vcHost)
-			if err != nil {
-				return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
-					"failed to get vCenter. Error: %+v", err)
+			var filteredDatastores []*cnsvsphere.DatastoreInfo
+			for _, datastores := range fsEnabledClusterToDsInfoMap {
+				filteredDatastores = append(filteredDatastores, datastores...)
 			}
-			volumeID, faultType, err = common.CreateFileVolumeUtil(ctx, cnstypes.CnsClusterFlavorVanilla,
-				vc, c.managers.VolumeManagers[vcHost], c.managers.CnsConfig, &createVolumeSpec,
-				filteredDatastores, filterSuspendedDatastores, false, nil)
-			if err != nil {
-				return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
-					"failed to create volume. Error: %+v", err)
+			if len(filteredDatastores) == 0 {
+				// when len(filteredDatastore)==0, it means vsan file service is not enabled on any vsan cluster
+				return nil, csifault.CSIVSanFileServiceDisabledFault, logger.LogNewErrorCode(log, codes.FailedPrecondition,
+					"no datastores found to create file volume, vSAN file service may be disabled")
 			}
-		} else {
-			volumeID, faultType, err = common.CreateFileVolumeUtil(ctx, cnstypes.CnsClusterFlavorVanilla,
-				vcenter, c.manager.VolumeManager, c.manager.CnsConfig, &createVolumeSpec,
-				filteredDatastores, filterSuspendedDatastores, false, nil)
-			if err != nil {
-				return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
-					"failed to create volume. Error: %+v", err)
+
+			if multivCenterCSITopologyEnabled {
+				// Get VC instance.
+				vcenter, err = common.GetVCenterFromVCHost(ctx, c.managers.VcenterManager, vcHost)
+				if err != nil {
+					return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
+						"failed to get vCenter instance for host %q. Error: %+v", vcHost, err)
+				}
+				volumeID, faultType, err = common.CreateFileVolumeUtil(ctx, cnstypes.CnsClusterFlavorVanilla,
+					vcenter, c.managers.VolumeManagers[vcHost], c.managers.CnsConfig, &createVolumeSpec,
+					filteredDatastores, filterSuspendedDatastores, false, nil)
+				if err != nil {
+					return nil, faultType, logger.LogNewErrorCodef(log, codes.Internal,
+						"failed to create volume. Error: %+v", err)
+				}
+			} else {
+				vcenter, err = c.manager.VcenterManager.GetVirtualCenter(ctx, c.manager.VcenterConfig.Host)
```

Comment:
I remember you recommended to not use this way of getting vcenter?

Reply:
When multi-VC FSS is disabled, this was the way to get VC object.